packages/ember-simple-auth/tests/helpers/start-app.js

```diff
@@ -1,12 +1,11 @@
 import Application from '../../app';
 import config from '../../config/environment';
-import { merge } from '@ember/polyfills';
 import { run } from '@ember/runloop';
 
 export default function startApp(attrs) {
-  let attributes = merge({}, config.APP);
+  let attributes = Object.assign({}, config.APP);
```

Comment:
Can we use `assign` from `@ember/polyfills` here as well?

Reply:
Is that needed in test land? I see that `Object.assign` is used in other test files as well:
https://github.com/simplabs/ember-simple-auth/search?q=Object.assign

Reply:
Yeah that was my earlier point – we should actually never just use it in Ember code (`.eslintrc.js` etc. is fine). The current usages should be considered a bug really.

Reply:
@marcoow Isn't it an option to let consumers bring an actual `Object.assign` polyfill? I would guess they are using that already anyways if they still support IE11 in 2022.

Using the ember assign polyfill means that 4.0 apps will show that deprecation warning until ember-simple-auth officially drops 3.x support, which doesn't seem ideal.

Alternatively, I think it should be possible to use `@embroider/macros` and only use Ember's assign if the consuming app is on a 3.x release.

Reply:
Sure but it we're already cleaning things up (and will be using `assign` from `@ember/polyfill` in other places still anyway) we might just as well update the existing usages?

> Using the ember assign polyfill means that 4.0 apps will show that deprecation warning until ember-simple-auth officially drops 3.x support, which doesn't seem ideal.

I don't think that's a problem tbh and we won't drop support for 3.x for quite some time. Aiming for 0 deprecations at all times isn't something anyone should be doing…

Reply:
We could also do something like this:

```js
const assign = Object.assign || emberAssign;
```

So similar to what was done before when falling back to `merge`.
This won't trigger the deprecation in Ember v4.

Reply:
I'd maybe have a util module in ESA for that so we don't have to repeat that everywhere in the app 👍